prov/util/src/util_coll.c

```diff
@@ -764,6 +745,43 @@ static struct fi_ops util_coll_fi_ops = {
 	.ops_open = fi_no_ops_open,
 };
 
+static inline int util_coll_mc_alloc(struct util_coll_mc **coll_mc)
+{
+	*coll_mc = calloc(1, sizeof(**coll_mc));
+	if (!*coll_mc)
+		return -FI_ENOMEM;
+
+	(*coll_mc)->mc_fid.fid.fclass = FI_CLASS_MC;
+	(*coll_mc)->mc_fid.fid.context = NULL;
+	(*coll_mc)->mc_fid.fid.ops = &util_coll_fi_ops;
+	(*coll_mc)->mc_fid.fi_addr = (uintptr_t)* coll_mc;
+
```

Comment:
Here we should save the av_set and ep, and bump reference counts on both.  The references should be dropped prior to freeing the util_coll_mc.